src/main/java/org/openstreetmap/atlas/checks/flag/serializer/CheckFlagDeserializer.java

```diff
@@ -51,15 +96,303 @@ public CheckFlag deserialize(final JsonElement json, final Type typeOfT,
+                        // current length of the geometry
+                        case UPDATE:
+                            final List<Location> updateLocations = this.convertWktToLocations(
+                                    descriptor.get(AFTER_VIEW).getAsString());
+                            for (int index = 0; index < updateLocations.size(); index++)
+                            {
+                                // If this is an existing position, replace it
+                                if (newGeometry.size() > position + index)
+                                {
+                                    newGeometry.set(position + index, updateLocations.get(index));
+                                }
+                                // Else add to the end
+                                else
+                                {
+                                    newGeometry.add(updateLocations.get(index));
+                                }
+                            }
+                            break;
+                        default:
+                            break;
+                    }
+                });
+
+        afterView.withGeometry(newGeometry);
+    }
+
+    /**
+     * Apply a relation member change descriptor to a CompleteEntity.
+     *
+     * @param afterView
+     *            {@link CompleteEntity}
+     * @param descriptor
+     *            {@link JsonObject} relation member change descriptor
+     */
+    private void applyRelationMemberChange(final CompleteEntity afterView,
+            final JsonObject descriptor)
+    {
+        // Only apply to relations
+        if (afterView instanceof CompleteRelation)
+        {
+            // Generate a bean item form the description
+            final RelationBean.RelationBeanItem member = new RelationBean.RelationBeanItem(
+                    descriptor.get(ID).getAsLong(), descriptor.get(ROLE).getAsString(),
+                    ItemType.valueOf(descriptor.get(ITEM_TYPE).getAsString().toUpperCase()));
+
+            // Get the existing relation bean. A RelationBean must be used to apply the change
+            // because the bounds of the member cannot be automatically calculated
+            final RelationBean bean = ((CompleteRelation) afterView).getBean();
+
+            switch (ChangeDescriptorType
+                    .valueOf(descriptor.get(GeoJsonConstants.TYPE).getAsString().toUpperCase()))
+            {
+                // Add the new member to the bean and add the bean to the relation
+                case ADD:
+                    bean.add(member);
+                    ((CompleteRelation) afterView).withMembers(bean,
+                            ((CompleteRelation) afterView).bounds());
+                    break;
+                // Remove the member from the bean and add the bean to the relation
+                case REMOVE:
+                    bean.removeItem(member);
+                    ((CompleteRelation) afterView).withMembers(bean,
```

Comment:
It could have some side effects to downstream processes if they assume that there aren't pointless relations with no members, but the check flag that gets created will still be valid.